adafruit_thermal_printer/thermal_printer.py

```diff
@@ -287,15 +287,17 @@ def reset(self):
         # ESC + 'D' + tab stop value list ending with null to terminate.
         self.send_command("\x1BD\x04\x08\x10\x14\x18\x1C\x00")
 
-    def print(self, text, end="\n"):
+    def print(self, text, *, end="\n", encoding="utf-8"):
         """Print a line of text.  Optionally specify the end keyword to
         override the new line printed after the text (set to None to disable
-        the new line entirely).
+        the new line entirely). Optionally specify the encoding. Some 
+        printers only accept the more restrictive encodings "cp437" and 
+        "ascii".
```

Comment:
Can you add a note here that encodings other than `"utf-8"` are only accepted on non-microcontroller platforms?

Reply:
I don't exactly understand what you mean. In my opinion, the encoding is not depending on the micro-controller, but on the printer that is used. In the project I did, I used a Raspberry Pi that can handle any encoding, but the problem arose from the printer that used the very specific encoding "cp437".
Could you elaborate on that?

Reply:
The Raspberry Pi can handle other encodings, but CircuitPython, which runs on the microcontroller targets, can only do UTF-8 encodings.  The issue is the the microcontrollers are unable to use any other encoding, so that should be the default and we should note that in the docstring here - that using anything else on those devices isn't supported.